makelib/golang.mk

```diff
@@ -30,10 +30,6 @@ GO_SUBDIRS ?= cmd pkg
 # Optional. Additional subdirs used for integration or e2e testings
 GO_INTEGRATION_TESTS_SUBDIRS ?=
 
-# Optional directories (relative to CURDIR)
-GO_VENDOR_DIR ?= vendor
```

Comment:
We don't vendor if GO111MODULE is on. `GO111MODULE` is set to `on` automatically if there is a `go.mod` file since Go 1.16: https://go.dev/blog/go116-module-changes